src/parsers/cdm_parser.jl

```diff
@@ -752,30 +724,90 @@ function make_storage(data::PowerSystemRaw, gen, bus)
     return battery
 end
 
-function parse_forecast_data_files(data::PowerSystemRaw)
+function parse_forecast_data_files(sys::System, data::PowerSystemRaw)
     forecast_data = ForecastInfos()
+    label_cache = Dict()
 
     for forecast in iterate_rows(data, TIMESERIES_POINTERS::InputCategory)
         simulation = forecast.simulation
         category = _get_component_type_from_category(forecast.category)
-        component_name = forecast.component_name
-        label = forecast.label
+        component = get_forecast_component(sys, category, forecast.component_name)
+        label, per_unit = _get_label_info!(label_cache, data, typeof(component), forecast)
         data_file = forecast.data_file
-        add_forecast_data!(forecast_data, simulation, category, component_name, label,
+        add_forecast_data!(forecast_data, simulation, component, label, per_unit,
                            joinpath(data.directory, data_file))
     end
 
     return forecast_data
 end
 
-const CATEGORY_TO_TYPE = Dict{String, DataType}(
+"""Return the forecast label and whether to convert to per_unit from the descriptor."""
+function _get_label_info!(label_cache::Dict, data::PowerSystemRaw, component_type, forecast)
+    if forecast.label_source == "Category"
+        if component_type <: Generator
+            category = GENERATOR::InputCategory
+        elseif component_type <: Service
+            category = RESERVES::InputCategory
+        elseif component_type <: Bus
```

Comment:
We shouldn't have forecasts for buses.

Reply:
It's a bit confusing.  The timeseries_pointers.csv file indicates that LoadZone timeseries data comes from the MW Load column of a bus.

I'm open to walking through this in person.

Reply:
I guess this is ok as long as we are clear that this applied to the CDM data format only. I am begging to worry we are making a lot of code generalizations using the specific CDM data as the bases.

Reply:
Keep in mind that this code is a CDM-specific implementation for parsing raw timeseries data into PowerSystems forecasts.  The code in forecast_parser.jl is supposed to apply to all data sources.